app/src/main/java/com/compscieddy/meetinthemiddle/GroupAdapter.java

```diff
@@ -14,40 +14,55 @@
  */
 public class GroupAdapter extends RecyclerView.Adapter<GroupAdapter.GroupHolder> {
 
-    private Context mContext;
+  private Context mContext;
+  private static ClickListener mClickListener;
 
-    @Override
-    public GroupAdapter.GroupHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        mContext = parent.getContext();
-        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
-        View view = layoutInflater.inflate(R.layout.list_group_item, parent, false);
-        return new GroupHolder(view);
-    }
+  public interface ClickListener{
+    void OnItemClick(View v);
+  }
 
-    @Override
-    public void onBindViewHolder(GroupAdapter.GroupHolder holder, int position) {
-        //Placeholder text for now
-        holder.mAvatarImageView.setImageResource(R.drawable.ic_account_circle_grey600_48dp);
-        holder.mTitleTextView.setText("Group " + position);
-        holder.mLastMessageTextView.setText("Last message of group " + position);
-    }
+  public void setClickListener(ClickListener clickListener){
+    mClickListener = clickListener;
+  }
 
-    @Override
-    public int getItemCount() {
-        //Arbitrary for now
-        return 10;
+  @Override
+  public GroupAdapter.GroupHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+    mContext = parent.getContext();
+    LayoutInflater layoutInflater = LayoutInflater.from(mContext);
+    View view = layoutInflater.inflate(R.layout.list_group_item, parent, false);
+    return new GroupHolder(view);
+  }
+
+  @Override
+  public void onBindViewHolder(GroupAdapter.GroupHolder holder, int position) {
+    //Placeholder text for now
+    holder.mAvatarImageView.setImageResource(R.drawable.ic_account_circle_grey600_48dp);
+    holder.mTitleTextView.setText("Group " + position);
+    holder.mLastMessageTextView.setText("Last message of group " + position);
+  }
+
+  @Override
+  public int getItemCount() {
+    //Arbitrary for now
+    return 10;
+  }
+
+  public static final class GroupHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+    CircleImageView mAvatarImageView;
+    TextView mTitleTextView;
+    TextView mLastMessageTextView;
+
+    public GroupHolder(View itemView) {
+      super(itemView);
+      mAvatarImageView = (CircleImageView) itemView.findViewById(R.id.group_avatar_image_view);
```

Comment:
Would rename to avatarImageView without the "m" prefix - the m prefix is nice when the file is larger and more complex, but I think it's actually a little unnecessary here for a small contained Holder class such as this